uamqp/message.py

```diff
@@ -131,12 +192,41 @@ def __str__(self):
             return ""
         return str(self._body)
 
-    def _parse_message(self, message):
+    def _parse_message_properties(self):
```

Comment:
We could probably improve things further be breaking this down to only load the specific properties on demand (e.g. only delivery annotations) - however I think this is a good approach for an incremental change given then time constraints.

Reply:
yes, I agree. when I implement this, i thought we could be doing parse for each property without parsing them all.

Reply:
Aside from lazy loading, let's try to tune the parsing code itself in the next preview.

Reply:
It occurs to me that doesn't the EH consumer _always_ use the message properties to get the offset while it's running?
In which case isn't the lazy loading somewhat ignored?
Or does the consumer not pull the annotations for _every_ message?

Reply:
aha, you got the point here!

Previously we update consumer.offset every time we create an event from a message. But I find that we don't need to update the consumer.offset such frequently. We just need to update the offset according to the last offset.

So when we receive 100 msgs, we just need the offset information of the last msg, not each one.

https://github.com/Azure/azure-sdk-for-python/blob/647b2561318ae127870b0ca503b09b5fbe09a7f9/sdk/eventhub/azure-eventhubs/azure/eventhub/aio/consumer_async.py#L203-L209